### PR TITLE
executor, infoschema, util: add sum_errors and sum_warnings to statement summary tables

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve?

Problem Summary:
Add `sum_errors` and `sum_warnings` to `statements_summary`.
These 2 fields indicate the total errors and warnings encountered for each kind of SQL.

### What is changed and how it works?

What's Changed:
ONLY summarize runtime errors now. It's a big move to also consider compiler errors, so it's not suitable for 4.0.
MySQL: Statements that has parser errors are not summarized in statement summary tables.
TiDB: Statements that has parser errors or compiler errors are not summarized.

How it Works:
Error: check whether it succeeded.
Warning: get `StmtCtx.WarningCount()`.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

N/A

### Release note 

- Add two fields `sum_errors` and `sum_warnings` to statement summary tables.

### CODE REVIEW

File: infoschema/tables_test.go [hunk 1]
```diff
@@ -1201,3 +1201,28 @@ func (s *testTableSuite) TestStmtSummaryInternalQuery(c *C) {
 		"1 select original_sql , bind_sql , default_db , status , create_time , update_time , charset , collation from mysql . bind_info" +
 			" where update_time > ? order by update_time"))
 }
+
+// Test error count and warning count.
+func (s *testTableSuite) TestStmtSummaryErrorCount(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
```
Comment (AilinKid) [inline infoschema/tables_test.go:1207]: tk.MustExec("use test")

File: infoschema/tables_test.go [hunk 2]
```diff
@@ -1201,3 +1201,28 @@ func (s *testTableSuite) TestStmtSummaryInternalQuery(c *C) {
...
+
+	// clear summaries
```
Comment (AilinKid) [inline infoschema/tables_test.go:1209]:
```suggestion
	// Clear summaries.
```

File: util/stmtsummary/statement_summary.go [hunk 3]
```diff
@@ -609,6 +612,10 @@ func (ssElement *stmtSummaryByDigestElement) add(sei *StmtExecInfo, intervalSeco
 	// refreshInterval may change anytime, update endTime ASAP.
 	ssElement.endTime = ssElement.beginTime + intervalSeconds
 	ssElement.execCount++
+	if !sei.Succeed {
+		ssElement.sumErrors += 1
+	}
+	ssElement.sumWarnings += int(sei.StmtCtx.WarningCount())
```
Comment (Deardrops) [inline util/stmtsummary/statement_summary.go:618]: Do there has data race?

**Pull request merged**